File: 1678-Goal Parser Interpretation-设计 Goal 解析器/1678-Goal Parser Interpretation-设计 Goal 解析器.c
char* interpret(char* command) {
	int len = strlen(command);
	char* ret = (char*)malloc((len + 1) * sizeof(char));
	int j = 0;
	for (int i = 0; i < len; i++)
	{
		if (command[i] == 'G')
			ret[j++] = command[i];
		else if (command[i] == '(')
		{
			if (command[i + 1] == ')')
			{
				ret[j++] = 'o';
				i++;
			}
			else
			{
				ret[j++] = 'a';
				ret[j++] = 'l';
				i += 3;
			}
		}
	}
	ret[j] = '\0';
	return ret;
}

File: 1678-Goal Parser Interpretation-设计 Goal 解析器/1678-Goal Parser Interpretation-设计 Goal 解析器.js
/**
 * @param {string} command
 * @return {string}
 */
var interpret = function(command) {
    let res = ''
    for(let i = 0, len = command.length; i < len; i++){
        switch(command[i]){
            case 'G':
                res += 'G'
                break
            case '(':
                if(command[i+1] === 'a'){
                    i++
                    res += 'al'
                }else if(command[i+1] === ')'){
                    res += 'o'
                }
                break
        }
    }
    return res
};
